airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -0,0 +1,240 @@
+        # check for errors and status, return when COMPLETED.
+        if not self.job_check_for_errors(response):
+            status = response.json().get("data", {}).get("node", {}).get("status")
+            if status == ShopifyBulkStatus.COMPLETED.value:
+                self.log_job_status(bulk_job_id, status)
+                return response.json().get("data", {}).get("node", {}).get("url")
+            elif status == ShopifyBulkStatus.RUNNING.value:
+                self.log_job_status(bulk_job_id, status)
+                # wait for the `job_check_interval_sec` value in sec before check again.
+                sleep(self.job_check_interval_sec)
+                return self.job_check_status(url, bulk_job_id)
+            elif status == ShopifyBulkStatus.FAILED.value:
+                self.log_job_status(bulk_job_id, status)
+                raise ShopifyBulkExceptions.BulkJobFailed(
+                    f"The BULK Job: `{bulk_job_id}` exited with {status}, details: {self.job_check_for_errors(response)}",
+                )
+            elif status == ShopifyBulkStatus.TIMEOUT.value:
+                self.log_job_status(bulk_job_id, status)
+                raise ShopifyBulkExceptions.BulkJobTimout(
+                    f"The BULK Job: `{bulk_job_id}` exited with {status}, please retry the operation with smaller date range.",
+                )
+            elif status == ShopifyBulkStatus.ACCESS_DENIED.value:
+                self.log_job_status(bulk_job_id, status)
+                raise ShopifyBulkExceptions.BulkJobAccessDenied(
+                    f"The BULK Job: `{bulk_job_id}` exited with {status}, please check your PERMISSION to fetch the data for this stream.",
+                )
+            else:
+                raise Exception(f"Could not validate the status of the BULK Job `{bulk_job_id}`.")
+
+    def job_check(self, url: str, created_job_response: requests.Response) -> Optional[str]:
+        """
+        This method checks the status for the BULK Job created, using it's `ID`.
+        The time spent for the Job execution is tracked to understand the effort.
+        """
+        bulk_job_id: str = self.job_get_id(created_job_response)
+        if bulk_job_id:
+            job_started = time()
+            try:
+                return self.job_check_status(url, bulk_job_id)
+            except (
+                ShopifyBulkExceptions.BulkJobFailed,
+                ShopifyBulkExceptions.BulkJobTimout,
+                ShopifyBulkExceptions.BulkJobAccessDenied,
+            ) as bulk_job_error:
+                raise bulk_job_error
+            except Exception as e:
+                raise ShopifyBulkExceptions.BulkJobUnknownError(f"The BULK Job: `{bulk_job_id}` has unknown status. Trace: {repr(e)}.")
+            finally:
+                time_elapsed = round((time() - job_started), 3)
+                self.logger.info(f"The BULK Job: `{bulk_job_id}` time elapsed: {time_elapsed} sec.")
+        else:
```

Comment:
We should not, since we have the `concurrency` job checks, and `job_get_id` will return `None` explicitly. In this case, we will `pass` until the `max_concurrency_attempts` value is reached, or when the `job` is created.

Reply:
can you add a comment explaining why a response might not have a job id and that we'll keep trying if that happens

Reply:
Added, typical scenario:  Job  was not created because of the `concurrent` request